packages/ui/src/components/Node/TableNodeComponent.tsx

```diff
@@ -184,8 +157,24 @@ const TableNodeComponent = ({ id, data }: {
     }
 
     setLoading(false);
-  }
+    setTotal(fetchedTotal);
+  }, [id, loading, offset, server, total]);
+
+  const dataStoryEvent = useCallback((event: DataStoryEventType) => {
+    if (event.type === DataStoryEvents.RUN_START) {
+      setItems([])
+      setOffset(0);
+      setIsRunning(false);
+    }
+
+    if (event.type === DataStoryEvents.RUN_SUCCESS) {
+      loadTableData();
+      setIsRunning(true);
+    }
+  }, []);
 
+  useDataStoryEvent(dataStoryEvent);
+  useIntersectionObserver(loaderRef.current!, loadTableData);
```

Comment:
Im not following this implementation 100%. Can you please update PR description:
* remove outdated info/pics
* add some background on this implementation
* resummarize what has been done and how (other things)

Reply:
Thanks for your thorough and patient review 👍. The latest info has been updated in the `PR` description.